### PR TITLE
Chunked encoding for _cat APIs

### PR DESCRIPTION
**Pull request by DaveCTurner**
Some of these APIs scale as O(#indices) or O(#shards) so it seems worthwhile to use chunked encoding throughout.

Relates #89838

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/rest/action/cat/RestTable.java [hunk 1]
```diff
@@ -57,60 +68,145 @@ private static XContentType getResponseContentType(RestRequest request) {
...
+                        writer.flush();
+                    } else {
+                        writer.close();
+                    }
+
+                    final var chunkOutput = currentOutput;
+                    currentOutput = null;
+                    return new ReleasableBytesReference(chunkOutput.bytes(), () -> Releasables.closeExpectNoException(chunkOutput));
+                } finally {
+                    if (currentOutput != null) {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/rest/action/cat/RestTable.java:194]: Although I don't see any obvious ways to throw exceptions in this area, I'm somewhat dubious about how such an exception would be handled in the networking code.

**Comment by DaveCTurner** [comment]
@elasticmachine update branch

**Review by original-brownbear — COMMENTED** [review]
Looks great thanks! Just one question on a test.

File: test/framework/src/main/java/org/elasticsearch/rest/RestResponseUtils.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    private RestResponseUtils() {}
+
+    private static final int PAGE_SIZE = 1 << 14;
+
+    public static BytesReference getBodyContent(RestResponse restResponse) throws IOException {
+        if (restResponse.isChunked() == false) {
+            return restResponse.content();
+        }
+
+        final var recycler = new Recycler<BytesRef>() {
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply test/framework/src/main/java/org/elasticsearch/rest/RestResponseUtils.java:28]: Ah yeah I extracted this from the other `new Recycler` bit above (with the smaller pages and extra checks) but you're right it doesn't do anything the non-recycled implementation does any more.

**Review by original-brownbear — APPROVED** [review]
LGTM!

**Pull request merged**